src/sbml/SBMLErrorLog.cpp

```diff
@@ -417,6 +427,104 @@ SBMLErrorLog::getErrorWithSeverity(unsigned int n, unsigned int severity) const
 
 
 /** @cond doxygenIgnored */
+
+LIBSBML_EXTERN
+SBMLErrorLog_t*
+SBMLErrorLog_create(void)
+{
+  return new(nothrow) SBMLErrorLog;
+}
+
+
+LIBSBML_EXTERN
+void
+SBMLErrorLog_free(SBMLErrorLog_t* log)
+{
+  delete log;
+}
+
+
+LIBSBML_EXTERN
+void
+SBMLErrorLog_add(SBMLErrorLog_t *log, const SBMLError_t *error)
+{
+  if (log == NULL || error == NULL) return;
+  log->add(*error);
+}
+
+
+LIBSBML_EXTERN
+const SBMLError_t *
+SBMLErrorLog_getError(const SBMLErrorLog_t *log, unsigned int n)
+{
+  if (log == NULL) 
+    return NULL;
+
+  return (const SBMLError_t*)(XMLErrorLog_getError((const XMLErrorLog_t*)(log), n));

```

Comment:
I think we should still return a dynamic cast to an SBMLError, since the error returned might not be an SBMLError. At least we should have a test for this.

Reply:
ok reverted my static casts